packages/cli/src/reporters/json.ts

```diff
@@ -0,0 +1,95 @@
+import * as fs from 'fs'
+import * as path from 'path'
+
+import AbstractListReporter, { checkFilesMap } from './abstract-list'
+import { CheckRunId } from '../services/abstract-check-runner'
+import { formatDuration, printLn, getTestSessionUrl } from './util'
+
+const outputFile = './checkly-json-report.json'
+
+type JsonBuilderOptions = {
+  testSessionId?: string
+  numChecks: number
+  runLocation: string
+  checkFilesMap: checkFilesMap
+}
+
+export class JsonBuilder {
+  testSessionId?: string
+  numChecks: number
+  runLocation: string
+  checkFilesMap: checkFilesMap
+  hasFilenames: boolean
+
+  constructor (options: JsonBuilderOptions) {
+    this.testSessionId = options.testSessionId
+    this.numChecks = options.numChecks
+    this.runLocation = options.runLocation
+    this.checkFilesMap = options.checkFilesMap
+    this.hasFilenames = !(options.checkFilesMap.size === 1 && options.checkFilesMap.has(undefined))
+  }
+
+  render (): string {
+    const testSessionSummary: any = {
+      testSessionId: this.testSessionId,
+      numChecks: this.numChecks,
+      runLocation: this.runLocation,
+      checks: [],
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const [_, checkMap] of this.checkFilesMap.entries()) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      for (const [_, { result, testResultId }] of checkMap.entries()) {
```

Comment:
The reporter currently doesn't handle timeouts so well. Running `npx checkly test --reporter=json --timeout=1` I get:

```
{
  "numChecks": 2,
  "runLocation": "eu-west-1",
  "checks": [
    {
      "result": "Fail",
      "name": "homepage.spec.ts",
      "duration": "NaNs",
      "filename": "__checks__/homepage.spec.ts",
      "link": null
    }
  ]
}
```

I believe we have the timeout error in `result.runError` and just need to handle that.

Reply:
will check

Reply:
@clample I just handled this by reporting `null` if no duration is recorded for the check run. Seems like the most simple way to solve this

Reply:
@tnolet avoiding the empty duration is good, but we also need to make sure the user knows that the check timed out. Otherwise I think users will be confused about what happened. I think we should have the timeout error (I believe it's in `result.runError`) somewhere in the output.

Reply:
@clample ah yes, this works:

```
   {
      "result": "Fail",
      "name": "local.spec.ts",
      "durationMilliseconds": null,
      "filename": "__checks__/local.spec.ts",
      "link": "https://app.checklyhq.com/test-sessions/a33d6e8a-34c5-44a6-9ba2-01219d44847c/results/2fe21340-dc32-44f8-b65e-67869e6e11e9",
      "runError": "Reached timeout of 1 seconds waiting for check result."
    },
```